main.cpp

```diff
@@ -56,51 +56,66 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
+    T(int inValue, const char *inName) : value(inValue), name(inName) {}   //1
     //2
+    int value;
     //3
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+struct Comparator                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T *compare(T *a, T *b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr) 
+        {
+            if (a->value < b->value) return a;
+            if (a->value > b->value) return b;        
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float first{0}, second{0};
+
+    float update(float *updatedValue)      //12
     {
-        
+        std::cout << "U's first value: " << first << std::endl;
+        first = *updatedValue;
+        std::cout << "U's first updated value: " << first << std::endl;
+        while (std::abs(second - first) > 0.001f)
+        {
+            second += 0.00001f;
+        }
+        std::cout << "U's second updated value: " << second << std::endl;
+        return second * first;
     }
 };
 
-struct <#structname2#>
+struct Updater
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float update(U *that, float *updatedValue)        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's first value: " << that->first << std::endl;
+        that->first = *updatedValue;
+        std::cout << "U's first updated value: " << that->first << std::endl;
+        while (std::abs(that->second - that->first) > 0.001f)
```

Comment:
What do we always do before using pointers?
also, put the `*` next to the type name, not the variable name.
it makes it more clear that it's a pointer.   This also aligns with the JUCE Coding Standard
- [ ] revise